### PR TITLE
fix limit condition

### LINKED ISSUE #50: Limit not working properly
image

If the limit is below 12, the function returns an empty list, for anything between 12-24 it returns 12 and so on.

The solution is simply changing the '>=' to '>' in:
https://github.com/mtshrmn/horrible-downloader/blob/eb6bec5321de6ba5f25e05663d357185c8183f1e/HorribleDownloader/parser.py#L87

I will submit a pull request but wanted to get feedback first in case I'm missing something.

Python 3.8.2, latest version of the package

### PR DESCRIPTION
**Pull request by Gbd199**
#50

### CODE REVIEW

**Comment by Gbd199** [comment]
I had to adjust the tests to match the new limitation checking.
In the original one, a limit of 3 would return an empty list and that's just wrong. If the user wants 3 episodes he shall get 3 episodes.
0 is an edge case, where the _get_html method will return 12 and not 0 but imo adding a special check for 0 is an overkill(in the end _get_uris returns 0 anyways).
I also added a proper limitation check of how many episodes does the get_episodes() method returns instead of just how many the internal _get_html() does. It just makes more sense since this is what the user ends up using and getting.

File: HorribleDownloader/parser.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def _get_html(showid: int, limit: int, show_type: str) -> str:
             response = requests.get(api, params=query)
             # the limit is counted in number of episodes (or batches)
             # because each page contains 12 episodes, we must divide it by 12.
-            if response.text == stop_text or query["nextid"] >= limit // 12:
+            if response.text == stop_text or query["nextid"] > limit // 12:
```
Comment (mtshrmn) [inline HorribleDownloader/parser.py]: I would change it to `(limit - 1) // 12` because then it wont send another request. I might be wrong as this was not tested.

**Review by mtshrmn — COMMENTED** [review]
Everything seems good, except of the one comment I've added. I don't have time to test it myself so if you can do it yourself I'll be very glad.

File: HorribleDownloader/parser.py (hunk 1 above)
Comment (Gbd199) [reply HorribleDownloader/parser.py]: pretty smart, it handles the 0 case, and all the other limits still work too. I updated the tests again to match this change.

**Comment by mtshrmn** [comment]
This seems good, thank you for adding the extra test on the proper limitation check.
